lib/lsp-data/standard_nums.rb

```diff
@@ -0,0 +1,209 @@
+# frozen_string_literal: true
+
+require 'library_stdnums'
+module LspData
+  ### Get all library standard numbers from a MARC record;
+  ###   default to requiring the OCLC prefix
+  def standard_nums(record:, input_prefix: true, output_prefix: false)
+    lccn = lccns(record)
+    isbn = isbns(record)
+    issn = issns(record)
+    oclc = oclcs(record: record,
+                 input_prefix: input_prefix,
+                 output_prefix: output_prefix)
+    { lccn: lccn, isbn: isbn, issn: issn, oclc: oclc }
+  end
+
+  def lccns(record)
+    lccn = []
+    f010 = record.fields('010').select { |f| f['a'] }
+    f010.each do |field|
+      value = StdNum::LCCN.normalize(field['a'])
+      lccn << value if value
+    end
+    lccn.uniq
+  end
+
+  def isbns(record)
+    isbn = []
+    f020 = record.fields('020').select { |f| f['a'] }
+    f020.each do |field|
+      value = isbn_normalize(field['a'])
+      isbn << value if value
+    end
+    isbn.uniq
+  end
+
+  ### Convert ISBN-10 to ISBN-13
+  def isbn10to13(isbn)
+    stem = isbn[0..8]
+    return nil if stem =~ /\D/
+
+    existing_check = isbn[9]
+    return nil if existing_check && existing_check != checkdigit_isbn10(stem)
+
+    main = ISBN13PREFIX + stem
+    checkdigit = checkdigit_isbn13(main)
+    main + checkdigit
+  end
+
+  ### Calculate check digit for ISBN-10
+  def checkdigit_isbn10(stem)
+    int_index = 0
+    int_sum = 0
+    stem.each_char do |digit|
+      int_sum += digit.to_i * (10 - int_index)
+      int_index += 1
+    end
```

Comment:
Another potential approach:

```suggestion
    int_sum = stem.chars
      .each_with_index
      .reduce(0) {|int_sum, (digit, int_index)| int_sum += digit.to_i * (10 - int_index) }
```